clouddq/runners/dbt/dbt_runner.py

```diff
@@ -213,15 +218,33 @@ def _resolve_dbt_path(
             logger.debug(f"Using 'dbt_path': {dbt_path}")
         return dbt_path
 
-    def _prepare_dbt_project_path(self) -> None:
+    def _prepare_dbt_project_path(self,
+                                  intermediate_table_expiration_hours: Optional[int]) -> None:
+
         dbt_project_path = self.dbt_path.absolute().joinpath("dbt_project.yml")
         if not dbt_project_path.is_file():
             logger.debug(
                 f"Cannot find `dbt_project.yml` in path: {dbt_project_path} \n"
                 f"Writing templated file to: {dbt_project_path}/dbt_project.yml"
             )
-            write_templated_file_to_path(dbt_project_path, DBT_TEMPLATED_FILE_LOCATIONS)
-        logger.debug(f"Using 'dbt_project_path': {dbt_project_path}")
+            if intermediate_table_expiration_hours:
+                set_dbt_intermediate_expiration_hours(
+                    path=dbt_project_path,
+                    intermediate_table_expiration_hours=intermediate_table_expiration_hours,
+                    lookup_table=DBT_TEMPLATED_FILE_LOCATIONS
+                )
```

Comment:
It seems like this code snippet will run regardless of whether `dbt_project_path` is a file or not.

Can we refactor this code a little bit?

```
if intermediate_table_expiration_hours:
... ...
else:
    if dbt_project_path.is_file():
        logger.debug(f"Using 'dbt_project_path': {dbt_project_path}")
    else:
        logger.debug(...)
        write_templated_file_to_path(dbt_project_path, DBT_TEMPLATED_FILE_LOCATIONS)
```

Reply:
refractored